internal/experimental/registry/util.go

```diff
@@ -54,3 +55,28 @@ func ctx(out io.Writer, debug bool) context.Context {
 	orascontext.GetLogger(ctx).Logger.SetLevel(logrus.DebugLevel)
 	return ctx
 }
+
+// parseReference will parse and validate the reference, and clean tags when
+// applicable tags are only cleaned when plus (+) signs are present, and are
+// converted to underscores (_) before pushing
+// See https://github.com/helm/helm/issues/10166
+func parseReference(raw string) (registry.Reference, error) {
+	// The sole possible reference modification is replacing plus (+) signs
+	// present in tags with underscores (_). To do this properly, we first
+	// need to identify a tag, and then pass it on to the reference parser
+	// NOTE: Passing immediately to the reference parser will fail since (+)
+	// signs are an invalid tag character, and simply replacing all plus (+)
+	// occurrences could invalidate other portions of the URI
+	parts := strings.Split(raw, ":")
+	if len(parts) > 1 && !strings.Contains(parts[len(parts)-1], "/") {
+		tag := parts[len(parts)-1]
+
+		if tag != "" {
+			// Replace any plus (+) signs with known underscore (_) conversion
+			newTag := strings.ReplaceAll(tag, "+", "_")
+			raw = strings.ReplaceAll(raw, tag, newTag)
```

Comment:
If the path happens to contain the same string as the old tag it would be replaced and broken. I will admit this is an incredibly rare situation (if it were to ever happen). I'm not even sure a registry would handle a `+` in their namespace. Just making the observation.

Reply:
you're right, but if we're assuming this only works for registries adhering with the oci distribution spec, then we should be covered by the limitations on registry namespaces ([here](https://github.com/opencontainers/distribution-spec/blob/main/spec.md#pulling-manifests)).

buuut that also doesn't excuse the fact that there isn't a check for the scenario you outlined

Reply:
Thanks for filling in details on the namespace spec.